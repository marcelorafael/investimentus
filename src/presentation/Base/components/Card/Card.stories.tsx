import React from 'react'
import { withKnobs } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react-native'
import { View, Text } from 'react-native'
import { Card } from './Card'

storiesOf('Base/Card', module)
  .addDecorator(withKnobs)
  .add('default', () => (
    <Card width={156} height={156} background="#F5F5F5">
      <View style={{ flex: 1 }}>
        <Text style={{ color: '#FFF' }}>Teste</Text>
      </View>
    </Card>
  ))
