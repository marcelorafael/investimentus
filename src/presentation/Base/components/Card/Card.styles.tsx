import styled from 'styled-components/native'

interface WrapperProps {
  width: number | string
  height: number
  background: string
}

export const Wrapper = styled.View<WrapperProps>`
  border-radius: ${(props) => props?.theme?.SIZES?.radius ?? 12}px;
  background-color: ${(props) => props.background};
  padding: 20px;
  margin: 5px;

  height: ${(props) => props.height}px;
  width: ${(props) => props.width};
`
