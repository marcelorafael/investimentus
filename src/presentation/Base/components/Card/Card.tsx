import React from 'react'
import { StyleSheet } from 'react-native'
import { Wrapper } from './Card.styles'
export type TypeDimensions = number | string

interface CardProps {
  width: number | string
  height: number
  background: string
  children: React.ReactElement
}

const Card: React.FC<CardProps> = (props: CardProps) => {
  const { width, height, background, children } = props
  return (
    <Wrapper
      width={width}
      height={height}
      background={background}
      style={{ ...styles.shadow }}
    >
      {children}
    </Wrapper>
  )
}

const CardMemo = React.memo(Card)

export { CardMemo as Card }

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center'
  },
  shadow: {
    shadowColor: '#222',
    shadowOffset: {
      width: 0,
      height: 4
    },
    shadowOpacity: 0.3,
    shadowRadius: 4.65,

    elevation: 8,
    marginHorizontal: 5
  }
})
