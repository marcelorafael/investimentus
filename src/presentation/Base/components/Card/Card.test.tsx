import React from 'react'

import renderer from 'react-test-renderer'
import { Card } from './Card'

describe('Base/Card', () => {
  it('should match snapshot', () => {
    const sut = (
      <Card background="#000" height={50} width={50}>
        <></>
      </Card>
    )

    const render = renderer.create(sut)

    expect(render).toMatchSnapshot()
  })
})
