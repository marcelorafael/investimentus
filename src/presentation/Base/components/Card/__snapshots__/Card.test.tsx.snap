// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base/Card should match snapshot 1`] = `
<View
  background="#000"
  height={50}
  style={
    Array [
      Object {
        "backgroundColor": "#000",
        "borderRadius": 12,
        "height": 50,
        "marginBottom": 5,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
        "width": 50,
      },
      Object {
        "elevation": 8,
        "marginHorizontal": 5,
        "shadowColor": "#222",
        "shadowOffset": Object {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 4.65,
      },
    ]
  }
  width={50}
/>
`
